streams/src/test/java/org/apache/kafka/streams/processor/internals/ProcessorStateManagerTest.java

```diff
@@ -509,4 +556,33 @@ public void shouldDeleteCheckpointFileOnCreationIfEosEnabled() throws Exception
         }
     }
 
+    private ProcessorStateManager getStandByStateManager(TaskId taskId) throws IOException {
+        return new ProcessorStateManager(
+            taskId,
+            noPartitions,
+            true,
+            stateDirectory,
+            new HashMap<String, String>() {
+                {
+                    put(persistentStoreName, persistentStoreTopicName);
+                }
+            },
+            changelogReader,
+            false);
+    }
+
+    private MockStateStoreSupplier.MockStateStore getPersistentStore() {
+        return new MockStateStoreSupplier.MockStateStore("persistentStore", true);
+    }
+
+    private static class MockBatchingRestoreCallback extends AbstractNotifyingBatchingRestoreCallback {
```

Comment:
Collapsed the mock objects in `CompositeRestoreListenerTest` and `ProcessorStateManagerTest` into a single class in `o.a.k.test.MockBatchingStateRestoreListener`, but we still need the `MockNoListenBatchingStateRestoreCallback` as we need a callback that does not implement the `StateRestoreListener` for testing.